### PR TITLE
Feature/blacksmith

### PR DESCRIPTION
**Pull request by Raycoms**
# Changes proposed in this pull request:
- Adds a Blacksmith worker to create tools, weapons and armor
- Adds the Stonemason worker to fulfill stone recipes
- Abstract the Sawmill code to use it in the classes of the above two
- Fix issue with dman picking up the items the crafters need

Review please

### CODE REVIEW

**Review by marchermans — DISMISSED** [review]

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 1]
```diff
@@ -1,47 +1,54 @@
...
 {
     /**
-     * Description of the job executed in the hut.
+     * Description string of the building.
      */
     private static final String STONEMASON = "Stonemason";
 
     /**
-     * Description of the block used to set this block.
+     * The min percentage something has to have out of wood to be craftable by this worker.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: old sawmill comment? :D

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 2]
```diff
@@ -1,47 +1,54 @@
...
+import net.minecraft.item.ItemStack;
 import net.minecraft.util.math.BlockPos;
+import net.minecraftforge.oredict.OreDictionary;
 import org.jetbrains.annotations.NotNull;
 
+import static com.minecolonies.api.util.constant.BuildingConstants.CONST_DEFAULT_MAX_BUILDING_LEVEL;
+
 /**
- * Creates a new building for the stonemason.
+ * Class of the sawmill building.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: old sawmill comment

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 3]
```diff
@@ -52,15 +59,14 @@ public String getSchematicName()
...
+        return CONST_DEFAULT_MAX_BUILDING_LEVEL;
     }
 
-    //TODO Implement Later
     @NotNull
     @Override
     public AbstractJob createJob(final CitizenData citizen)
     {
-        return new JobPlaceholder(citizen);
+        return new JobSawmill(citizen);
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: wrong job?

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 4]
```diff
@@ -70,16 +76,65 @@ public String getJobName()
...
+        int amountOfValidBlocks = 0;
+        int blocks = 0;
+        for(final ItemStack stack : storage.getInput())
+        {
+            if(!ItemStackUtils.isEmpty(stack))
+            {
+                if (stack.getItem() instanceof ItemBlock)
+                {
+                    final Block block = ((ItemBlock) stack.getItem()).getBlock();
+                    if (block == Blocks.STONEBRICK || block == Blocks.STONE_BRICK_STAIRS || block == Blocks.STONE_SLAB || block == Blocks.STONE_SLAB2)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java:103]: intended that its stone/stonebrick only? or is cobblestone missing

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 5]
```diff
@@ -1,47 +1,54 @@
...
+    private static final double MIN_PERCENTAGE_TO_CRAFT = 0.75;
 
     /**
-     * Max building level of the hut.
-     */
-    private static final int MAX_BUILDING_LEVEL = 3;
-
-    /**
-     * Instantiates the building.
+     * Instantiates a new sawmill building.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: stonemason :)

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 6]
```diff
@@ -70,16 +76,65 @@ public String getJobName()
...
+                blocks++;
+            }
+        }
+
+        return amountOfValidBlocks > 0 && blocks/amountOfValidBlocks > MIN_PERCENTAGE_TO_CRAFT;
+    }
+
     /**
-     * ClientSide representation of the building.
+     * BuildingDeliveryman View.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: stonemason xD

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java [hunk 7]
```diff
@@ -70,16 +76,65 @@ public String getJobName()
...
      */
-    public static class View extends AbstractBuildingWorker.View
+    public static class View extends AbstractBuildingCrafter.View
     {
+
         /**
-         * Instantiates the view of the building.
+         * Instantiate the deliveryman view.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java]: stonemason aswell^^

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/stonemason/EntityAIWorkStonemason.java [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+ * Crafts stone related block when needed.
+ */
+public class EntityAIWorkStonemason extends AbstractEntityAICrafting<JobStonemason>
+{
+    /**
+     * Initialize the sawmill and add all his tasks.
+     *
+     * @param sawmill the job he has.
+     */
+    public EntityAIWorkStonemason(@NotNull final JobStonemason sawmill)
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/stonemason/EntityAIWorkStonemason.java]: parameter name

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/stonemason/EntityAIWorkStonemason.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import com.minecolonies.coremod.entity.ai.basic.AbstractEntityAICrafting;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * Crafts stone related block when needed.
+ */
+public class EntityAIWorkStonemason extends AbstractEntityAICrafting<JobStonemason>
+{
+    /**
+     * Initialize the sawmill and add all his tasks.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/stonemason/EntityAIWorkStonemason.java]: stonemason :D

File: src/main/java/com/minecolonies/coremod/entity/ai/citizen/stonemason/EntityAIWorkStonemason.java [hunk 10]
```diff
@@ -0,0 +1,22 @@
...
+     *
+     * @param sawmill the job he has.
```
Comment (someaddons) [inline src/main/java/com/minecolonies/coremod/entity/ai/citizen/stonemason/EntityAIWorkStonemason.java]: stonemason^^

**Review by someaddons — COMMENTED** [review]
looks fine overall, just found some descriptions off and one wrong job :)

File: src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java (hunk 4 above)
Comment (Raycoms) [reply src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingStonemason.java:103]: cobblestone and stone have oredictionary ids

**Review by someaddons — DISMISSED** [review]

**Review by marchermans — DISMISSED** [review]

**Pull request merged**